### PR TITLE
MFB-1191: Federal medicare savings size parameter update and NC Medicare Savings program update

### PR DESCRIPTION
**Pull request by myang1010**
## Context & Motivation

1. Confirmed the Federal MSP eligibility rule for size parameter should be the couple/individual.
2. Updated NC MSP program to adopt the two steps check on eligibility

- Linear: [MFB-1911](https://linear.app/myfriendben/issue/MFB-1191/federalmedicare-savings-couple-scoped-income-tested-against-whole)

## Changes Made

- programs/programs/federal/medicare_savings/calculator.py, use couple/individual
- programs/programs/nc/medicaid/nc_medicare_savings/calculator.py, use two steps eligibility check
- programs/programs/nc/medicaid/nc_medicare_savings/tests/test_nc_msp.py, add tests

## Testing

- Migrations to run: none
- Configuration updates needed: none
- Environment variables/settings to add: none
- Manual testing steps: 

```
 Scenario 1: Medicaid Individual - expected PASS
 1 person, age 68, Medicare coverage
 SS Retirement income: $1,000/month
 Assets: $5,000
 
 Scenario 2: Medicaid Couple - expected PASS
 Head: age 67, Medicare, SS Retirement $900/month
 Spouse: age 65, Medicare, SS Retirement $700/month
 Assets: $8,000
 
 Scenario 3: Individual with Ineligible Spouse, no dependents - expected FAIL
 Head: age 66, Medicare, SS Retirement $800/month (unearned)
 Spouse: age 50, no Medicare, wages $3,000/month (earned)
 
 Scenario 4: Individual with Ineligible Spouse + teen child - expected PASS
 Head: age 66, Medicare, wages $1,500/month
 Spouse: age 55, no Medicare, wages $1,500/month
 Child: age 14
 
 Scenario 5: Individual with Ineligible Spouse + dep child - expected PASS
 Head: age 66, Medicare, SSRI $1,500/month
 Spouse: age 55, no Medicare/health insurance, wages $1,500/month 
 Child: age 15
 
 Scenario 6: Individual with SSI Spouse + dep child - expected PASS
 Head: age 66, Medicare, wages $2,800/month
 Spouse: age 55, Medicaid, SSI $93/mo
 Child: age 14

```

## Deployment

None

## Notes for Reviewers

- Here's a [GoogleDoc](https://docs.google.com/document/d/1NB7AFQSbqDGCIJLsU4ucBxastTc42F3Qd8sL4nx69l4/edit?tab=t.0) me and Louise used for this NC MSP program change in case it helps. 

## Summary by CodeRabbit

* **Eligibility Updates**
  * Updated North Carolina Medicare Savings eligibility calculations to account for household assets, marital status, dependents, SSI, and deemed income.
  * Added an alternate family-size methodology for applicants who do not qualify under SSI budgeting rules.
  * Updated North Carolina asset thresholds and income calculations.

* **Bug Fixes**
  * Federal poverty-level household size is now calculated from marital status when evaluating Medicare Savings eligibility.

* **Tests**
  * Added coverage for individual, couple, spouse, dependent-child, SSI, income, and asset eligibility scenarios.

### CODE REVIEW

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 1]
```diff
@@ -6,50 +6,211 @@
...
+        capped_size = min(family_size, 8)
+        _, max_income = self.get_fpl_limits(capped_size)
+        if max_income is None:
+            return None
+        if family_size > 8:
+            max_income += ADDITIONAL_PER_PERSON * (family_size - 8)
 
-        e.condition(min_income is not None and countable_income >= min_income)
-        e.condition(max_income is not None and countable_income <= max_income)
+        return max_income
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py:226]: Fixed.

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 2]
```diff
@@ -6,50 +6,211 @@
...
+        ]
+        # Skip step 2 entirely if no dependents under 18 in the home
+        if not dependents:
+            return False
 
-        # compute combined income (excluding SSI)
-        earned_total = 0
-        unearned_total = 0
-        for m in included_members:
+        family_size = 1 + (1 if spouse else 0) + len(dependents)
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py:193]: Every member has a real PK by the time this code runs. None pk won't occur.

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 3]
```diff
@@ -6,50 +6,211 @@
...
+            )
+            return "medicaid_individual" if parents_on_ssi else "medicaid_child"
+        if not spouse:
+            return "medicaid_individual"
+        if spouse.calc_gross_income("yearly", ["sSI"]) > 0:
+            return "medicaid_individual"  # spouse already on SSI
+        spouse_medicare_eligible = spouse.has_insurance_types(("medicare",), strict=False) or spouse.age >= 65
+        if spouse_medicare_eligible:
+            return "medicaid_couple"
+        return "individual_with_ineligible_spouse"
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py]: Fixed.

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 4]
```diff
@@ -6,50 +6,211 @@
...
+
+        fpl = self.program.year.as_dict().get(household_size)
+        if fpl is None:
+            return False
+
+        if countable_income > fpl:
+            return False
+        # Income <= 100% FPL: only qualify for MSP if assets disqualify from full Medicaid
+        status = "married" if spouse else "single"
+        return self.screen.household_assets > self.medicaid_asset_limit[status]
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py:120]: We'll keep this check here.

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 5]
```diff
@@ -7,50 +7,216 @@ class MedicareSavingsNC(MedicareSavings):
...
+        e.condition(
+            member.age >= self.min_age
+            or member.has_insurance_types(("medicare",), strict=False)
+            or member.calc_gross_income("yearly", ["sSDisability"]) > 0
+        )
 
-        # insurance
+        # Asset limit
+        status = "married" if spouse else "single"
+        e.condition(self.screen.household_assets <= self.asset_limit[status])
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py:41]: Fixed.

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 6]
```diff
@@ -7,50 +7,216 @@ class MedicareSavingsNC(MedicareSavings):
...
+        unearned = other_member.calc_gross_income("yearly", ["unearned"], ["sSI"])
+
+        if unearned >= self.living_allowance:
+            unearned -= self.living_allowance
+        else:
+            remaining = self.living_allowance - unearned
+            unearned = 0
+            earned = max(0, earned - remaining)
+
+        return max(0, earned + unearned)
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py]: Here's the deeming rule, 
```
For deeming cases only (Individual w/ Ineligible Spouse / Medicaid Child):
1.Calculate a/b's own income with exclusions above
2. Separately calculate ineligible spouse's/parent's income:
Subtract $498 Living Allowance from their unearned income first, then any remainder from earned income
3.Compare what's left after the allowance to the Living Allowance limit:
If remainder is <= limit: don't deem anything - a/b is evaluated on their own income only
4. If remainder is > limit: add only the excess to a/b's income
The $498 Living Allowance must be updated as a class variable every year (will be NCDHHS Medicaid for Adults Manual) 
```
No plan to change.

[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py]: Fixed the Step 3.

File: programs/programs/nc/medicaid/nc_medicare_savings/calculator.py [hunk 7]
```diff
@@ -6,51 +6,221 @@
...
+            unearned -= self.living_allowance
+        else:
+            remaining = self.living_allowance - unearned
+            unearned = 0
+            earned = max(0, earned - remaining)
+
+        remainder = max(0, earned + unearned)
+        if remainder <= self.living_allowance:
+            return 0.0
+        return remainder
```
[the remark this answers is not in the document]
Comment (myang1010) [reply programs/programs/nc/medicaid/nc_medicare_savings/calculator.py:165]: No. The MA-2260 language says "deem the excess" where excess means what's left after the living allowance deduction - that's already remainder. The living allowance has already been subtracted once to produce remainder.

return remainder - self.living_allowance would subtract it a second time, effectively applying a double-deduction for high-income deemors.

return remainder is correct.

File: programs/programs/cross_white_label/msp/nc.py [hunk 8]
```diff
@@ -6,51 +6,221 @@
...
+        """
+        They qualify if any of these are true:
+            Reports Medicare as their health insurance
+            Age > 65 (regardless of insurance)
+            Reports SSDI income (regardless of age)
+        Automatic disqualifiers:
+            Reports SSI income (they get full Medicaid instead)
+        """
         member = e.member
-        super().member_eligible(e)
```
Comment (nihalerdal) [inline programs/programs/cross_white_label/msp/nc.py:17]: `member_eligible` no longer calls `super().member_eligible(e)`, which also removes the base class insurance whitelist check:

```
e.condition(member.insurance.has_insurance_types(self.eligible_insurance_types))
# eligible_insurance_types = ("none", "employer", "private", "medicare")
```
Just to make sure I understand, was this change intentional? Are none/employer/private/medicare still valid insurance types for this program?

File: programs/programs/cross_white_label/msp/nc.py [hunk 9]
```diff
@@ -6,51 +6,221 @@
...
+            earned, unearned = self.apply_income_disregards(earned, unearned)
+            countable_income = earned + unearned
+            household_size = 1
+
+        elif classification == "medicaid_couple":
+            earned, unearned, _ = self.get_combined_income(member, spouse, include_ssi=False)
+            earned, unearned = self.apply_income_disregards(earned, unearned)
+            countable_income = earned + unearned
+            household_size = 2
+
```
Comment (nihalerdal) [inline programs/programs/cross_white_label/msp/nc.py:96]: There is some repeated logic across these four classification blocks. I was wondering if it'd be worth considering consolidating the common parts to reduce the duplication. I can also see a readability tradeoff since the current cases are more self-contained, so I just wanted to open this up for discussion.

File: programs/programs/cross_white_label/msp/nc.py [hunk 10]
```diff
@@ -6,51 +6,221 @@
...
+        Step 2: Court-ordered family size - applicant + spouse + dependents under 18, no deeming.
+        Eligible if either step passes.
+        """
+
+        e.condition(self._passes_ssi_budgeting(member, spouse) or self._passes_family_size_methodology(member, spouse))
+
+    def _passes_ssi_budgeting(self, member, spouse) -> bool:
+        """Step 1: individual/couple SSI budgeting, income <= 100% FPL."""
+
         """
```
Comment (nihalerdal) [inline programs/programs/cross_white_label/msp/nc.py:63]: Tiny nit: There is a second `"""` block here after the function docstring. It looks like this is an unused string literal. Maybe it would be cleaner to merge them into the main docstring.

**Review by nihalerdal — COMMENTED** [review]
Overall looks good, Ming! I found the changes pretty clear. I added one question regarding the insurance logic. The rest are just minor/nit suggestions. Let me know what you think!

**Review by SonaliBedge — COMMENTED** [review]
Nice work Ming! I tested all the manual test scenarios you mentioned in the PR description on my local, reviewed your changes, and ran the `test_nc.py` file. Everything looks good after addressing Nihal’s comments.

Small nitpicks:

* The `member_eligible` docstring says **"Age > 65"**, but the code checks **`>= 65`**.
* Optional: If possible, you could add a few more tests to cover edge cases in `test_nc.py`, such as:


  - Income at or below 100% FPL, assets at or below the Medicaid limit. For example, a single person with $1,000/mo income and $1,500 in assets should be ineligible, because they'd get full Medicaid instead. This covers the return self.screen.household_assets > self.medicaid_asset_limit[status] branch.
  - Deemed-income cutoff. An ineligible spouse whose income, after the living allowance, is at or below the allowance should deem $0 income. That's the remainder <= self.living_allowance path in _calc_deemed_income.
  - Family size over 8. An applicant with 8 or more kids under 18 exercises the additional_per_person math in _get_family_size_income_limit.
  - Spouse on SSI with a dependent. Check that the spouse's SSI doesn't count toward Step 2 income. Scenario 6 has this setup, but its income is too far below the limit for the test to notice.
  - Just over the limit. For example, Scenario 4 with income just above 135% FPL for 3 people should fail. That would confirm the limit is actually being applied, not just that the pass cases pass.
  - Benefit amount. Scenario 1's docstring says $2,436/year, but only eligible is asserted. One assertion on the value would lock in the $203/month premium.
